finagle-mysql/src/main/scala/com/twitter/finagle/mysql/Client.scala

```diff
@@ -33,6 +33,23 @@ object Client {
       }
     }
 
+    def transactional[R](f: Client => Future[R]): Future[R] = 
+      factory().flatMap { svc =>
+        val client = apply(ServiceFactory.const(svc))
```

Comment:
As @mosesn mentioned, we recently added this internally and it's available as part of the develop branch.

Note though, this may have surprising results. Because of how the mysql client operates on the this `ServiceFactory` you aren't guaranteed exclusivity to this `Service` (i.e. connection in this case). You need to override the `Service#close` method to completely control its lifecycle. Take a look at how we implemented it in the `develop` branch.